pocs/dome/astrohaven.py

```diff
@@ -148,14 +153,32 @@ def _full_move(self, send, target_feedback):
             True if the target_feedback is received from the dome before the MOVE_TIMEOUT;
             False otherwise.
         """
-        end_by = time.time() + AstrohavenDome.MOVE_TIMEOUT
-        while not self._nudge_shutter(send, target_feedback):
-            if time.time() < end_by:
-                continue
-            self.logger.error('Timed out moving the dome. Check for hardware or communications ' +
-                              'problem. send=%r latest_state=%r', send, self._read_latest_state())
-            return False
-        return True
+        # Set a short timeout on reading, so that we don't open or close slowly.
+        # In other words, we'll try to read status, but if it isn't available,
+        # we'll just send another command.
+        saved_timeout = self.serial.ser.timeout
```

Comment:
I wish `pyserial` supported timeouts on individual `read` and `write`.

Reply:
Ditto. I'll file an issue with pyserial.

Reply:
Filed https://github.com/pyserial/pyserial/issues/304